detection-rules/headers_replyto_mismatch_sus_tld.yml

```diff
@@ -0,0 +1,94 @@
+name: "Reply-to/Sender Mismatch with suspicious TLD"
+description: |
+  This rule detects a mismatch between the reply-to and the sender email addresses, and one or both of them are from suspicious TLDs
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and (
+    any(headers.reply_to,
+        .email.email != sender.email.email
```

Comment:
do we wanna check if the domains don't match too? that'd prevent us from firing on stuff like this:

From: sam@foo.agency
Reply-to: support@foo.agency